libfwupd/fwupd-enums.h

```diff
@@ -72,13 +72,14 @@ typedef enum {
  * FwupdDeviceFlags:
  * @FWUPD_DEVICE_FLAG_NONE:			No flags set
  * @FWUPD_DEVICE_FLAG_INTERNAL:			Device cannot be removed easily
- * @FWUPD_DEVICE_FLAG_ALLOW_ONLINE:		Is capable of 'live' updating
- * @FWUPD_DEVICE_FLAG_ALLOW_OFFLINE:		Is capable of 'offline' updating
+ * @FWUPD_DEVICE_FLAG_ALLOW_ONLINE:		Update can be called in the live desktop session
+ * @FWUPD_DEVICE_FLAG_ALLOW_OFFLINE:		Update can be called from offline mode
  * @FWUPD_DEVICE_FLAG_REQUIRE_AC:		Requires AC power
  * @FWUPD_DEVICE_FLAG_LOCKED:			Is locked and can be unlocked
  * @FWUPD_DEVICE_FLAG_SUPPORTED:		Is found in current metadata
  * @FWUPD_DEVICE_FLAG_NEEDS_BOOTLOADER:		Requires a bootloader mode
  * @FWUPD_DEVICE_FLAG_REGISTERED:		Has been registered with other plugins
+ * @FWUPD_DEVICE_FLAG_NEEDS_REBOOT:		Requires a reboot to apply firmware or to reload hardware
```

Comment:
Why not just get rid of `ALLOW_OFFLINE` and replace with `NEEDS_REBOOT ` entirely instead?  This would certainly simplify things.  

As of now I haven't ever seen https://www.freedesktop.org/software/systemd/man/systemd.offline-updates.html# in use for firmware updates.  I haven't personally envisioned any situation that you would rather schedule that for later on when the hardware might not be plugged in or might be a stateful update.

Looking at the plugins that use `ALLOW_OFFLINE `here's my thoughts for each if you drop the concept of offline updates in OS.

- **raspberry pi**; this shouldn't have an `FWUPD_DEVICE_FLAG_ALLOW_OFFLINE `set at all.  Just apply updates 
immediately.
- **colorhug**; this shouldn't have an `FWUPD_DEVICE_FLAG_ALLOW_OFFLINE `set at all.  Just apply updates 
immediately.
- **dell**; this could drop `FWUPD_DEVICE_FLAG_ALLOW_OFFLINE `in favor of `FWUPD_FEVICE_FLAG_NEEDS_REBOOT`
- **uefi**; this could drop `FWUPD_DEVICE_FLAG_ALLOW_OFFLINE `in favor of `FWUPD_FEVICE_FLAG_NEEDS_REBOOT`
colorhug; 

Another idea to fix the stateful hardware problem is to only offer offline when it's an internal device.